### ISSUE TITLE
Can create folders but not create files to /vagrant

### ISSUE THREAD

**Issue by howanu** [Parallels/vagrant-parallels#320]
Does anyone know how [issue 300](https://github.com/Parallels/vagrant-parallels/issues/300) was fixed? I'm seeing similar symptoms using

- Parallels 13.3.0
- Vagrant 2.0.3
- Mac High Sierra 10.13.4
- box based on bento/ubuntu-16.04

I can create directories on /vagrant, but not edit or create files.

> mount | grep vagrant
> vagrant on /vagrant type prl_fs (rw,relatime,sync)
> 

The same environment and box are working successfully for two colleagues.

Here's the issue:

> strace touch /vagrant/touch
execve("/usr/bin/touch", ["touch", "/vagrant/touch"], [/* 66 vars */]) = 0
brk(NULL)                               = 0x1314000
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=77660, ...}) = 0
mmap(NULL, 77660, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fcbc081a000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0P\t\2\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=1868984, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fcbc0819000
mmap(NULL, 3971488, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7fcbc023e000
mprotect(0x7fcbc03fe000, 2097152, PROT_NONE) = 0
mmap(0x7fcbc05fe000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1c0000) = 0x7fcbc05fe000
mmap(0x7fcbc0604000, 14752, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7fcbc0604000
close(3)                                = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fcbc0818000
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7fcbc0817000
arch_prctl(ARCH_SET_FS, 0x7fcbc0818700) = 0
mprotect(0x7fcbc05fe000, 16384, PROT_READ) = 0
mprotect(0x60e000, 4096, PROT_READ)     = 0
mprotect(0x7fcbc082d000, 4096, PROT_READ) = 0
munmap(0x7fcbc081a000, 77660)           = 0
brk(NULL)                               = 0x1314000
brk(0x1335000)                          = 0x1335000
open("/usr/lib/locale/locale-archive", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=2981280, ...}) = 0
mmap(NULL, 2981280, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fcbbff66000
close(3)                                = 0
open("/vagrant/touch", O_WRONLY|O_CREAT|O_NOCTTY|O_NONBLOCK, 0666) = -1 EACCES (Permission denied)
utimensat(AT_FDCWD, "/vagrant/touch", NULL, 0) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/locale.alias", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=2995, ...}) = 0
read(3, "# Locale name alias data base.\n#"..., 4096) = 2995
read(3, "", 4096)                       = 0
close(3)                                = 0
open("/usr/share/locale/en_US/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale-langpack/en_US/LC_MESSAGES/coreutils.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale-langpack/en/LC_MESSAGES/coreutils.mo", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=619, ...}) = 0
mmap(NULL, 619, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fcbc082c000
close(3)                                = 0
write(2, "touch: ", 7touch: )                  = 7
write(2, "cannot touch '/vagrant/touch'", 29cannot touch '/vagrant/touch') = 29
open("/usr/share/locale/en_US/LC_MESSAGES/libc.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale/en/LC_MESSAGES/libc.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale-langpack/en_US/LC_MESSAGES/libc.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/share/locale-langpack/en/LC_MESSAGES/libc.mo", O_RDONLY) = -1 ENOENT (No such file or directory)
write(2, ": Permission denied", 19: Permission denied)     = 19
write(2, "\n", 1
)                       = 1
close(1)                                = 0
close(2)                                = 0
exit_group(1)                           = ?
+++ exited with 1 +++

**Comment by romankulikov** [comment]
Can you please show `ls -ld` on the host directory shared with the virtual machine under the name "vagrant"?

**Comment by howanu** [comment]
Thanks for your response. The ls -ld shows permissions of "drwxrwxrwx".

However, I was able to fix the problem yesterday. I found that my host files' group ownership differed from that my colleagues. When I performed a chgrp -R to make our group ownerships match, the problem disappeared.

I'm still not sure why my default Mac group permissions would cause the problem. I've been using them for 3 years at least without any other types of permissions issues. At any rate, I'm unblocked now.
